[Lens] Help menu GH link and Visualization selection text

cchaos opened the pull request:
<img width="910" alt="Screenshot 2019-10-15 12 30 24" src="https://user-images.githubusercontent.com/666475/66851698-46bdb680-ef49-11e9-9521-8465ff382151.png">
<img width="556" alt="Screenshot 2019-10-15 11 55 03" src="https://user-images.githubusercontent.com/666475/66851700-49201080-ef49-11e9-92e2-20f8856a8852.png">



### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- ~[ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- ~[ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- ~[ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~
- ~[ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

elasticmachine commented:
Pinging @elastic/kibana-app

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/1222/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/1284/)

wylieconlon commented on src/legacy/core_plugins/kibana/public/visualize/wizard/type_selection/new_vis_help.tsx:
```diff
@@ -30,5 +30,16 @@ export const NewVisHelp = () => (
+    </p>
+    <EuiButton fill size="s" iconType="popout" iconSide="right">
+      Go to Lens
+    </EuiButton>
```
You're adding this to OSS code, which won't make sense in all cases Lens is not OSS. I think @chrisdavies might need to add to the visualization registry to support this

cchaos commented on x-pack/legacy/plugins/lens/public/help_menu_util.js:
```diff
@@ -0,0 +1,17 @@
+import { render, unmountComponentAtNode } from 'react-dom';
+import { HelpMenu } from './help_menu';
+
+export function addHelpMenuToAppChrome(chrome) {
```
Why remove the file when it just needs to be plugged into the header's help menu? As stated in the summary of the PR, I just need someone to do that part. It can be pushed directly to this branch.

cchaos commented:
@timroes @wylieconlon This is why I stated in the the summary of the PR that it needs to be "hooked up". I just added the text and button from a strictly UI perspective. Please feel free to update this branch directly.

cchaos commented on x-pack/legacy/plugins/lens/public/help_menu.js:
```diff
@@ -0,0 +1,40 @@
+    return (
+      <Fragment>
+        <EuiHorizontalRule margin="none" />
+        {/* Leaving the below in for future addition of docs link */}
```
The reason I left it in is so once the docs page was available, you wouldn't have to go hunt for the right UI to place the link as. As a compromise, I uncommented all the code and left the UI in but having it only render once the docs page is available via a variable.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/1583/)

wylieconlon commented on x-pack/legacy/plugins/lens/public/help_menu_util.tsx:
```diff
@@ -0,0 +1,48 @@
+      <EuiButton
+        fill
+        iconType="popout"
+        href={`${ELASTIC_WEBSITE_URL}guide/en/kibana/${DOC_LINK_VERSION}/lens.html`}
```
Is there a way to use the current language or kibana version?

wylieconlon commented on src/legacy/core_plugins/kibana/public/visualize/wizard/type_selection/new_vis_help.tsx:
```diff
@@ -19,16 +19,39 @@
+            <strong>{t.promotion!.description}</strong>
+          </p>
+          <EuiButton
+            href={chrome.addBasePath(t.aliasUrl)}
```
I think you could use the new platform `core.http` here? https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#client-side

wylieconlon commented on src/legacy/core_plugins/kibana/public/visualize/wizard/type_selection/type_selection.tsx:
```diff
@@ -152,7 +152,9 @@ class TypeSelection extends React.Component<TypeSelectionProps, TypeSelectionSta
                   <EuiSpacer size="m" />
-                  <NewVisHelp />
+                  <NewVisHelp
+                    promotedTypes={(visTypes as VisTypeAliasListEntry[]).filter(t => t.promotion)}
```
1. This could return multiple promotions?
2. This assumes the promotions are alias types, which I've commented on elsewhere

wylieconlon commented on src/legacy/core_plugins/kibana/public/visualize/wizard/type_selection/new_vis_help.tsx:
```diff
@@ -19,16 +19,39 @@
+            iconSide="right"
+          >
+            {t.promotion!.calloutText}
+          </EuiButton>
```
How easy is it to provide a link to one of the existing visualization types instead of an alias? If it's easy I'd say we should not assume that there is an `aliasUrl`

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/1842/)

cchaos commented:
@chrisdavies I just changed one thing from your new vis help stuff in which I changed `calloutText` to `buttonText` since that is more specific to the element.

I also assigned the rendering of the docs link to a variable since right now it leads to a 404. This way it's hidden until we actually have docs.

## Status

Handing off to devs.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/1865/)

AlonaNadler commented:
can we remove the arrow icon on Lens instead let's use a beta icon?
Lens being separate application than visualize isn't something that is crucial to mention 
![image](https://user-images.githubusercontent.com/32550316/66524824-66b42c80-eaa8-11e9-8447-684f502f6fc6.png)

wylieconlon commented:
Updated this to use a Beta icon

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/4746/)
* Commit: b1ec1a6bdb4ef80ebdea3679162500af794528e7

chrisdavies commented on x-pack/legacy/plugins/lens/public/help_menu_util.tsx:
```diff
@@ -0,0 +1,58 @@
+import { render, unmountComponentAtNode } from 'react-dom';
+import { Chrome } from 'ui/chrome';
+
+const docsPage = undefined;
```
The naming convention seems to be `lens.html` (e.g. `canvas.html` is canvas' landing page). So I suggest we make this `lens.html`, and if that turns out to be wrong, we can fix it later. If it turns out to be right, we're golden.

wylieconlon replied:
Since this page doesn't exist, I'm not sure we want to link to it already. But I see your point. This could be a thing we do later

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/4792/)
* Commit: 9691d01e3e36d65b7839c0c35d64ff3855d301ac
